src/main/scala/com/github/sbt/duplicates/ClasspathEntity.scala

```diff
@@ -0,0 +1,81 @@
+package com.github.sbt.duplicates
+
+import java.util.zip.CRC32
+
+import sbt._
+
+/**
+  * Lazy/Comparable handle to a class or resource within a zip file or directory.
+  * Caches the CRC32.
+  * Falls back to reading the full file contents if absolutely necessary.
+  */
+sealed trait ClasspathEntity {
+
+  /**
+    * Name of the zip file or directory.
+    */
+  def source: File
+
+  /**
+    * Path within the zip file or directory.
+    *
+    * @example `scala/None$.class`
+    * @example `META-INF/MANIFEST.MF`
+    */
+  def name: String
+
+  /**
+    * CRC32 hash as calculated for zip files.
+    * Lets us reuse the pre-calculated value in the jar/zip central directory.
+    *
+    * @see https://stackoverflow.com/q/10953958
+    */
+  def crc32: Int
+
+  /**
+    * The content of the class file or resource.
+    */
+  def bytes: Array[Byte]
+
+  override def equals(obj: Any): Boolean =
+    obj match {
+      case null => false
+      case other: ClasspathEntity =>
+        name == other.name && hashCode == other.hashCode && bytes.sameElements(other.bytes)
```

Comment:
Thinking further, the CRC32 comparison is adding less value than I thought. It only saves IO for the `ConflictState.ContentDiffer` case. That's the exceptional case rather than the norm since we're probbly running `checkDuplicatesTest` in a CI build. Over the common case, the crc32s will collide and we'll have to go to disk anyway.

[checksums.combinations(2).map{ c1 == c2 }](https://github.com/sbt/sbt-duplicates-finder/blob/3ba841dd40a0f61c1e8cf486b1708f305d54d742/src/main/scala/com/github/sbt/duplicates/Classpath.scala#L30-L32) perf could regress pathologically badly here if we have lots of duplicates with equal content.

I'm thinking we could replace the combinations with a `checksums.groupBy(_.bytes)` to get back to `O(n)` disk reads. Simple, but not memory-safe. To get memory safety back, we could either go back to a wide SHA (simple and probabilistically correct) or stream the content and compare bytewise (more complex, faster, 100% correct).

I'll play with this a bit more. Feedback welcome if you have a strong preference.

Reply:
I tossed the CRC32 and bytewise comparison and added back the original SHA-256. Seems like a good balance.

I may try to simplify further after dinner.

Reply:
Bug here. Should be comparing on `sha256`, not `hashCode`. I'll fix.